pkg/devfile/components.go

```diff
@@ -31,7 +32,7 @@ func GetKubernetesComponentsToPush(devfileObj parser.DevfileObj) ([]devfilev1.Co
 		componentName := ""
 		if command.Exec != nil {
 			componentName = command.Exec.Component
-		} else if command.Apply != nil {
+		} else if !allowApply && command.Apply != nil {
```

Comment:
I think it should be the opposite, or I don't understand the meaning of this option:

```suggestion
		} else if allowApply && command.Apply != nil {
```

Reply:
By default, this function ignores the K8s components referenced by any command.
1. It collects all the K8s component and puts it in a map of name:component.
2. Filter out the components referenced by any command(apply or exec) and delete them from the map. Here, if allowApply is true, `!allowApply & command.Apply!=nil` becomes false and the components referenced by Apply command will not be deleted from the map.
3. Return the list of remaining components, essentially the ones not referenced by any command.